packages/react-components/lib/place.ts

```diff
@@ -18,3 +29,38 @@ export function getPlaces(buildingMap: BuildingMap): Place[] {
   }
   return places;
 }
+
+export function getNamedPlaces(buildingMap: BuildingMap): NamedPlaces {
+  const places = new Map<string, Place>();
+  const pickupPoints = new Map<string, Place>();
+  const dropoffPoints = new Map<string, Place>();
+  const cleaningZones = new Map<string, Place>();
+
+  for (const level of buildingMap.levels) {
+    for (const graphs of level.nav_graphs) {
+      for (const vertex of graphs.vertices) {
+        if (vertex.name) {
+          const place: Place = { level: level.name, vertex };
+          for (const p of vertex.params) {
+            if (p.name === DEFAULT_PICKUP_POINT_PARAM_NAME) {
+              pickupPoints.set(vertex.name, place);
+            }
+            if (p.name === DEFAULT_DROPOFF_POINT_PARAM_NAME) {
+              dropoffPoints.set(vertex.name, place);
+            }
+            if (p.name === DEFAULT_CLEANING_ZONE_PARAM_NAME) {
+              cleaningZones.set(vertex.name, place);
+            }
+          }
+          places.set(vertex.name, place);
+        }
+      }
+    }
+  }
```

Comment:
https://github.com/open-rmf/rmf-web/pull/743/commits/9837319c40d7f40c6522e5c6dfe37bd9cd8481bf

I've just refactored the if case! Thanks for catching that, @Angatupyry!

Although the repeated if cases look ugly, I think I'll keep it that way since we have no guarantees that a single waypoint can't have multiple of these params (e.g. all 3), also I believe switch statements should be reserved for enums or simple number and chars, I don't think we should be comparing strings that way.

I'll go ahead and merge this now! Thanks!